src/index.js

```diff
@@ -98,6 +98,35 @@ export default function fetch(url, opts) {
 			finalize();
 		});
 
+		fixResponseChunkedTransferBadEnding(req, err => {
+			if (response.body.destroy) {
+				response.body.destroy(err);
+			} else {
+				// node < 8
+				response.body.emit('error', err);
+				response.body.end();
+			}
+		});
+
+		/* c8 ignore next 18 */
+		if (parseInt(process.version.substring(1)) < 14) {
+			// Before Node.js 14, pipeline() does not fully support async iterators and does not always
+			// properly handle when the socket close/end events are out of order.
+			req.on('socket', s => {
+				s.addListener('close', hadError => {
+					// if a data listener is still present we didn't end cleanly
+					const hasDataListener = s.listenerCount('data') > 0
```

Comment:
#1064 used the count of types of events with listeners registered on the socket to determine if the stream was closed prematurely but it's not reliable on node 8 and below, so instead if we've got a listener for `'data'` events we're still expecting data to arrive so use that to work out if we should emit the `ERR_STREAM_PREMATURE_CLOSE` error.